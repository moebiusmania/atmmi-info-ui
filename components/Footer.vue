<script lang="ts" setup>
const today: Date = new Date();
</script>

<template>
  <footer class="text-center p-6">
    <p class="font-light text-sm">
      <span>{{ today.getFullYear() }}</span> - Sviluppato da{{ " " }}
      <a
        href="https://github.com/moebiusmania/atmmi-info-ui"
        target="_blank"
        class="underline"
        rel="noopener noreferrer"
      >
        Salvatore Laisa
      </a>
    </p>
    <p class="font-light text-sm mt-3">
      Progetto indipendente e non affiliato o supportato da ATM Milano S.p.A.
    </p>
  </footer>
</template>
