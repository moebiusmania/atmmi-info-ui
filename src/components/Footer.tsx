
const Footer = (): JSX.Element => {
  const today = new Date();

  return (
    <footer className="text-center py-6">
      <p className="font-light text-sm">
        <span>{today.getFullYear()}</span> - Sviluppato da <a 
            href="https://github.com/moebiusmania/atmmi-info-ui" 
            target="_blank"
            className="underline" 
            rel="noopener noreferrer">
          Salvatore Laisa</a>
      </p>
      <p className="font-light text-sm mt-3">
        Progetto indipendente e non affiliato o supportato da ATM Milano S.p.A. 
      </p>
    </footer>
  )
};

export default Footer;