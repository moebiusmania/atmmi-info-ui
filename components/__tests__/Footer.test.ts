import { mount } from "@vue/test-utils";
import { expect, test } from "vitest";

import Footer from "../Footer.server.vue";

test("<Footer />", () => {
  const wrapper = mount(Footer);
  const year = new Date().getFullYear();

  expect(wrapper.html()).toMatchSnapshot();

  expect(wrapper.find("footer").exists()).toBe(true);
  expect(wrapper.text()).toContain(year);
});
