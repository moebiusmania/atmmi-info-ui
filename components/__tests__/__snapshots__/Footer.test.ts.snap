// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Footer /> 1`] = `
"<footer class="text-center p-6">
  <p class="font-light text-sm"><span>2025</span> - Sviluppato da <a href="https://github.com/moebiusmania/atmmi-info-ui" target="_blank" class="underline" rel="noopener noreferrer"> Salvatore Laisa </a></p>
  <p class="font-light text-sm mt-3"> Progetto indipendente e non affiliato o supportato da ATM Milano S.p.A. </p>
</footer>"
`;
